studio/pages/project/[ref]/functions/[functionSlug]/index.tsx

```diff
@@ -135,28 +222,140 @@ const PageLayout: NextPageWithLayout = () => {
+                    yAxisKey="count"
+                    stackKey="status"
+                    data={data}
+                    highlightedValue={sumBy(data, 'count')}
+                    customDateFormat={selectedInterval.format}
+                    stackColors={['brand', 'slate', 'yellow', 'red']}
+                    onBarClick={() => {
+                      router.push(
+                        `/project/${projectRef}/functions/${functionSlug}/invocations?its=${startDate.toISOString()}`
+                      )
+                    }}
+                  />
+                )
+              }}
+            />
+          ) : (
+            <ReportWidget
+              title="Invocations"
+              data={invocationsChartData}
+              isLoading={isInvStatsChartLoading}
+              renderer={(props) => (
+                <BarChart
+                  className="w-full"
+                  xAxisKey="timestamp"
+                  yAxisKey="count"
+                  data={props.data}
+                  highlightedValue={sumBy(props.data, 'count')}
+                  customDateFormat={selectedInterval.format}
+                  onBarClick={(v) => {
+                    router.push(
+                      `/project/${projectRef}/functions/${functionSlug}/invocations?its=${startDate.toISOString()}&ite=${
+                        v.timestamp
+                      }`
+                    )
+                  }}
+                />
+              )}
+            />
+          )}
+          {resourceUsageMetricsEnabled && (
+            <>
+              <ReportWidget
+                title="CPU time"
+                tooltip="Average CPU time usage for the function"
+                data={resourceChartData}
+                isLoading={isResourceUsageChartLoading}
+                renderer={(props) => {
+                  const latest = normalizedResourceUsageData[normalizedResourceUsageData.length - 1]
+                  let highlightedValue
+                  if (latest) {
+                    highlightedValue = latest['avg_cpu_time_used']
+                  }
+                  return (
+                    <AreaChart
+                      className="w-full"
+                      xAxisKey="timestamp"
+                      customDateFormat={selectedInterval.format}
+                      yAxisKey="avg_cpu_time_used"
+                      data={props.data}
+                      format="ms"
+                      highlightedValue={highlightedValue}
+                    />
                   )
                 }}
               />
-            )}
-          />
+              <ReportWidget
+                title="Memory"
+                tooltip="Average memory usage for the function"
+                data={resourceChartData}
+                isLoading={isResourceUsageChartLoading}
+                renderer={(props) => {
+                  const latest = normalizedResourceUsageData[normalizedResourceUsageData.length - 1]
```

Comment:
the renderer props contains the data passed into the ReportWidget in line 336. this allows us to decouple the renderer logic from the data fetching hooks.

Reply:
should use `props.data` instead of `normalizedResourceUsageData`.
Also, seems like there is some variable naming inconsistency, like `resourceChartData` vs `normalizedResourceUsageData`

Reply:
I thought this was done to get the last non-zero value. Anyway, updated to use `props.data`.